### PR TITLE
Add macOS build script

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by jcfr** [comment]
From @thewtex:  To help the binaries work on older versions of macOS, set [CMAKE_OSX_DEPLOYMENT_TARGET](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html) to `10.9`  as demonstrated in [ITKPythonPackage](https://github.com/InsightSoftwareConsortium/ITKPythonPackage/blob/b58c21e0e550cf55c7d0eb31f4fcb2e658bfdb27/scripts/macpython-build-wheels.sh#L100).

> I can also send a PR after this one to document the build process on OS X

@prabhuramachandran  Let's do this in a different PR after this one has been integrated.

Also note that the script was renamed from `osx_build_wheels.py` to `macos_build_wheels.py`

File: scripts/macos_build_wheel.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+def get_python_info():
+    py_exe = sys.executable
+    version = sys.version_info[:2]
+    py_ver = '{0}.{1}'.format(*version)
+    prefix = os.path.abspath(sys.prefix)
+    py_inc_dir = glob.glob(os.path.join(prefix, 'include', 'python*'))[0]
+    py_lib_dir = os.path.join(prefix, 'lib')
+    py_lib = os.path.join(py_lib_dir, 'libpython%s.dylib' % py_ver)
```
Comment (jcfr) [inline scripts/macos_build_wheel.py]: To ensure that the libraries are weekly linked, the path to the real library should NOT be used.

See https://github.com/InsightSoftwareConsortium/ITKPythonPackage/blob/b58c21e0e550cf55c7d0eb31f4fcb2e658bfdb27/scripts/macpython-build-common.sh#L33-L47

**Comment by prabhuramachandran** [comment]
@jcfr -- do I have push rights on this repo?  Do you want me to make any specific changes or are you going to?  Thanks for rebasing this on top of #14.

File: scripts/macos_build_wheel.py (hunk 1 above)
Comment (prabhuramachandran) [reply scripts/macos_build_wheel.py]: OK, so are you advocating a similar approach with a null directory on Linux and OS X?  Shouldn't this be needed on Linux also?

**Comment by jcfr** [comment]
> are you advocating a similar approach with an empty file on Linux and OS X?

yes 👍 

> Shouldn't this be needed on Linux also?

It is already the case

https://github.com/jcfr/VTKPythonPackage/blob/962ae4f89ca9b0c5cd5c46bdd8fd63eb31cb57f2/scripts/internal/manylinux-build-common.sh#L33-L47


> do I have push rights on this repo? 

Yes. I just added you as a collaborator

> Do you want me to make any specific changes or are you going to?

It would be great to implement something similar on macOS

**Comment by prabhuramachandran** [comment]
Perfect, thanks!  I'll make the changes later this week.  Spent too much midnight oil last few days.

**Comment by prabhuramachandran** [comment]
I've made the changes but am unable to push. Is there something special I should do to push, my remote looks like this:

upstream	git@github.com:jcfr/VTKPythonPackage.git (fetch)
upstream	git@github.com:jcfr/VTKPythonPackage.git (push)

but it does not allow me to push

**Comment by jcfr** [comment]
It looks like  you haven't accepted the invitation: https://github.com/jcfr/VTKPythonPackage/invitations

**Comment by prabhuramachandran** [comment]
Thanks!  Never seemed to have needed that before.

**Comment by prabhuramachandran** [comment]
OK I've pushed the suggested changes, please let me know if there are any other things I need to check -- I am re-running a build to test this out and it is churning away which is a good thing.

**Comment by jcfr** [comment]
Nitpick: Since commit entitled "Make suggested fixes as per review."  are not very informative, I would prefer you cherry pick the commit.

**Comment by prabhuramachandran** [comment]
OK I will send a more detailed commit message, is it OK if I do a force push to amend the commit?

**Comment by jcfr** [comment]
> is it OK if I do a force push to amend the commit

👍

**Comment by prabhuramachandran** [comment]
Hope this one works.

**Comment by prabhuramachandran** [comment]
Now that this is merged, maybe I'll update the docs over the weekend, OK?

**Comment by jcfr** [comment]
>  maybe I'll update the docs over the weekend, OK?

That would be great 👍

**Pull request merged**